### PR TITLE
feat: Allow Custom Compare Functions

### PR DESCRIPTION
**Pull request by davidfan168**
Most changes are copied from this pull request: https://github.com/praneshr/react-diff-viewer/pull/137/files

If a function is passed as `lineCompareMethod`, this function will be treated as the diff function. Users can pass other functions in the diff library, or pass their own functions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,11 +71,11 @@ class Diff extends PureComponent {
...
-| oldValue                  | `string                   | Object`                        | `''`                                                                                                                                                                                                                                                                                                                                                                                                             | Old value as string (or Object if using `diffJson`).                                                                                                                                                                                                                                                                                                                                                                                             |
```
Comment (davidfan168) [inline README.md:74]: The old readme was not showing up properly because `|` was treated as a table column

**Review by Aeolun — APPROVED** [review]
LGTM

**Comment by kevinlu1248** [comment]
@davidfan168 @Aeolun 

This change is actually wrong and causes a bug where the diff lines are sometimes hidden for some reason. I found that turning on dissabledWordDiff unhides them.

Instead, these are the lines you're supposed to change to change how the diff lines are handled:

https://github.com/Aeolun/react-diff-viewer-continued/blob/71d1374c3915b026739c52e780be06f40c48e7e3/src/compute-lines.ts#L137-L145

**Pull request merged**